lib/recode/formatter_plugin.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Recode.FormatterPlugin do
+  @moduledoc """
+  Defines Recode formatter plugin for `mix format`.
+
+  Since Elixir 1.13, it is possible to define custom formatter plugins. This
+  plugin allows you to run Recode autocorrecting tasks together when executing
+  `mix format`.
+
+  To use this formatter, simply add `Recode.FormatterPlugin` to your
+  `.formatter.exs` plugins:
+
+  ```
+    [
+      inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"],
+      plugins: [Recode.FormatterPlugin]
+    ]
+  ```
+
+  By default it uses the `.recode.exs` configuration file.
+
+  If your project does not have a `.recode.exs` configuration file, you can pass
+  the configuration using the `recode` option:
+
+  ```
+    [
+      inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"],
+      plugins: [Recode.FormatterPlugin],
+      recode: [
+        tasks: [
+          {Recode.Task.AliasExpansion, []},
+          {Recode.Task.EnforceLineLength, []},
+          {Recode.Task.SinglePipe, []}
+        ]
+      ]
+    ]
+  ```
+
+  """
+
+  @behaviour Mix.Tasks.Format
+
+  alias Recode.Config
+  alias Recode.Task
+  alias Rewrite.Project
+  alias Rewrite.Source
+
+  @impl true
+  def features(_opts) do
+    [extensions: [".ex", ".exs"]]
+  end
+
+  @impl true
+  def format(contents, opts) do
+    file = Keyword.fetch!(opts, :file)
+    config = Keyword.get(opts, :recode, [])
+    project = project_from_string(contents, file)
+
+    config
+    |> merge_project_config()
+    |> force_formatter_config(project)
+    |> Recode.Runner.run()
```

Comment:
I tried to keep the API changes minimal - but this might not be the best approach.

Perhaps `Recode.Runner.run/1` could be modified to make it also accept sources?